frontend/src/utils/menus.js

```diff
@@ -1,45 +1,53 @@
 import React from "react";
 import { Link } from "react-router-dom";
-import { Menu } from "antd";
 
-// renderMenuItem and matchingMenuKeys taken from chord_web
+/**
+ * @typedef {Required<import("antd").MenuProps>['items'][number]} MenuItem
+ * @typedef {Partial<{ key: string, icon: JSX.Element, text: string, url: string, onClick: () => void, children: BadMenuItem[], style: React.CSSProperties, disabled?: boolean }>} BadMenuItem
+ */
```

Comment:
this makes sense to me although i would suggest going for readability instead of overengineering, which seems to be a common outcome when you work on the frontend components. also formatting in vs code, it would be nice to not have the type definition in 1 line but in multiple lines since there are a lot of variables.

Reply:
I wasn't sure if multiple lines were possible in JSDoc. Turns out yes.